lib/Frontend/ModuleInterfaceBuilder.cpp

```diff
@@ -322,13 +329,23 @@ bool ImplicitModuleInterfaceBuilder::buildSwiftModuleInternal(
     SubError = (bool)subASTDelegate.runInSubCompilerInstance(
         moduleName, interfacePath, OutPath, diagnosticLoc,
         [&](SubCompilerInstanceInfo &info) {
-          auto EBuilder = ExplicitModuleInterfaceBuilder(
-              *info.Instance, diags, sourceMgr, moduleCachePath, backupInterfaceDir,
-              prebuiltCachePath, ABIDescriptorPath, extraDependencies, diagnosticLoc,
-              dependencyTracker);
-          return EBuilder.buildSwiftModuleFromInterface(
-              interfacePath, OutPath, ShouldSerializeDeps, ModuleBuffer,
-              CompiledCandidates, info.CompilerVersion);
+
+          auto build = [&] {
+            auto EBuilder = ExplicitModuleInterfaceBuilder(
+                *info.Instance, diags, sourceMgr, moduleCachePath, backupInterfaceDir,
+                prebuiltCachePath, ABIDescriptorPath, extraDependencies, diagnosticLoc,
+                dependencyTracker);
+            return EBuilder.buildSwiftModuleFromInterface(
+                interfacePath, OutPath, ShouldSerializeDeps, ModuleBuffer,
+                CompiledCandidates, info.CompilerVersion);
+          };
+
+          if (silenceInterfaceDiagnostics) {
+            ErrorDowngradeConsumerRAII R(*diags, /*Silence=*/true);
```

Comment:
Feel free to ignore this suggestion, but it might be nicer to avoid the lambda:
```c++
Optional<ErrorDowngradeConsumerRAII> R;
if (silenceInterfaceDiagnostics)
  R.emplace(*diags, /*Silence=*/true);
...
```

Reply:
Thank you for the suggestion, I ended up applying this logic with the local temporary DiagnosticEngine.